app/graphql/mutations/profile/associate_rules.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Mutations
+  module Profile
+    # Mutation to associate rules with a profile
+    class AssociateRules < BaseMutation
+      include ProfileHelper
+
+      graphql_name 'associateRules'
+
+      argument :id, ID, required: true
+      argument :rule_ids, [ID], required: true
+      field :profile, Types::Profile, null: true
+
+      def resolve(args = {})
+        profile = find_profile(args[:id])
+        if profile
+          rules_added, rules_removed = profile.update_rules(
+            ids: args[:rule_ids]
+          )
+          audit_mutation(profile, rules_added, rules_removed)
+        end
+        { profile: profile }
+      end
```

Comment:
Recommend wrapping this in a transaction

Reply:
It can be done in a followup PR.

Reply:
@akofink does this mean that we should have a transaction around the [associate hosts](https://github.com/RedHatInsights/compliance-backend/blob/master/app/graphql/mutations/profile/associate_systems.rb#L16) as well?

Reply:
It can't hurt! Thanks for noticing that.